x/supplier/types/message_unstake_supplier.go

```diff
@@ -6,15 +6,21 @@ const TypeMsgUnstakeSupplier = "unstake_supplier"
 
 var _ sdk.Msg = (*MsgUnstakeSupplier)(nil)
 
-func NewMsgUnstakeSupplier(address string) *MsgUnstakeSupplier {
+func NewMsgUnstakeSupplier(signerAddress, address string) *MsgUnstakeSupplier {
 	return &MsgUnstakeSupplier{
+		Signer:  signerAddress,
 		Address: address,
 	}
 }
 
 func (msg *MsgUnstakeSupplier) ValidateBasic() error {
```

Comment:
(same as above)

Reply:
When unstaking, we check that the signer is either the *supplier*'s owner or operator, which is not available in `ValidateBasic`. The `Address` here just indicate which operator to unstake.